erpnext/hr/doctype/leave_application/leave_application.py

```diff
@@ -258,10 +258,19 @@ def get_leave_balance(employee, leave_type, from_date, to_date):
 
 	leave_all = leave_all and flt(leave_all[0][0]) or 0
 
+	from erpnext.hr.report.employee_leave_balance.employee_leave_balance import get_leave_allocation_record
+	
+	la = get_leave_allocation_record(from_date)
+
 	leave_app = frappe.db.sql("""select SUM(total_leave_days)
 		from `tabLeave Application`
-		where employee = %s and leave_type = %s and to_date>=%s and from_date<=%s
-		and status="Approved" and docstatus = 1""", (employee, leave_type, from_date, to_date))
+		where employee = '%(employee)s' and leave_type = '%(leave_type)s' 
+			and (from_date between '%(from_date)s' and '%(to_date)s' 
+				or to_date between '%(from_date)s' and '%(to_date)s') 
+			and to_date < '%(to_date)s'
+		and status="Approved" and docstatus = 1"""% {"employee": employee, 
```

Comment:
This is not the right way to pass values to sql queries. You should use

`frappe.db.sql("some query", { "val1:" val1, "val2": val2 })`